lib/raph/parser/base_parser.rb

```diff
@@ -17,33 +17,20 @@ class BaseParser
       #    FileParser.new.id                  # => :files
       #    BaseArgumentParser.new.id          # => :base_arguments
       def id
-        name = class_name.gsub(/parser$/i, '')
-        name << 's' # make it plural
-        to_underscored_sym(name)
+        to_underscored_sym("#{self.class.name.sub(/^.*::/, '').sub(/parser$/i, '')}s")
       end
 
       # Parses arguments and returns results of parsing.
       def parse(args)
+        raise NotImplementedError
       end
 
-      protected
+      private
 
       # Returns underscored symbol of string
       # (snake case format).
       def to_underscored_sym(str)
-        str.gsub(/^[-]+/, '').
-          gsub(/([A-Z]+)([A-Z][a-z])/,'\1_\2').
-          gsub(/([a-z\d])([A-Z])/,'\1_\2').
-          tr("-", "_").
-          downcase.
-          to_sym
-      end
-
-      private
-
-      # Returns name of current class.
-      def class_name
-        self.class.name.split('::').last || ''
+        str.sub(/^-+/, '').gsub(/([A-Z\d]+)([A-Z][a-z])/, '\1_\2').gsub(/([a-z\d])([A-Z])/,'\1_\2').tr('-', '_').downcase.to_sym
```

Comment:
Sure, in general it's a matter of style, probably will be clearer to split.  
Thanks.